ScrapeRepos.ts

```diff
@@ -0,0 +1,356 @@
+const axios = require('axios');
+const cheerio = require('cheerio');
+const time = require('console');
+const showdown = require('showdown');
+
+
+/** Fetches the repo owner and name for each repo on the GitHub trending page
+ * @param {number} timeMode 0: daily; 1: weekly; 2: monthly
+ * @returns {string[]} an array that stores alternatingly the owner and the name of each repo
+ */
+async function fetchRepos( timeMode: number) {
+  try {
+    const timeModes = ['daily', 'weekly', 'monthly'];
+    const response = await axios.get('https://github.com/trending?since=' + timeModes[timeMode]);
+    const $ = cheerio.load(response.data);
+    const repos: string[] = [];
+
+    $('h2 a').each((i: number, el: cheerio.Element) => {
+      const repoName = $(el).text().trim();
+      repos.push(repoName);
+    });
+    
+    var trendingSplit: string[] = [];
+    // trim the repos to be correctly formatted
+    repos.forEach((repo) => {
+      let trimmedName = repo.replace(/\n\s+/g, '');
+      trimmedName = trimmedName.replace(/\//g, "");
+      const stringSplit = trimmedName.split(" ");
+      trendingSplit.push(stringSplit[0]);
+      trendingSplit.push(stringSplit[1]);
+    });
+
+    return trendingSplit;
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+
+/**  This function imports the ReadMe.md file for a repository (if it can be located)
+* @param {string} owner - owner of the repo
+* @param {string} name - name of the repo
+* @returns {string} a string containing the text of the repo. Returns null if the repo can't be located
+*/
+async function getReadme(owner: String, name: String) {
+  // these paths exists to check in multiple locations for the readme files
```

Comment:
Please add description on top of the function not inside the function if the description is not about something in the function